### ISSUE TITLE
Queue setup

### ISSUE THREAD

**Issue by NehzUx** [codalab/competitions-v2#630]
I follow the instruction here https://github.com/codalab/competitions-v2/wiki/Compute-Worker-Management---Setup to setup a new queue (not even a GPU one)

The command I used was: 

> docker run \
>     -v /var/run/docker.sock:/var/run/docker.sock \
>     --env BROKER_URL=pyamqp://dd64f051-6cd3-4c4e-a646-ef5980276e1f:1dc2fa65-fd93-4260-bab2-0f39d6917030@broker.codabench.org:9001/b348279e-acb4-4178-8778-a8805f7dda3c \
>     --name compute_worker \
>     -d \
>     --restart unless-stopped \
>     --log-opt max-size=1m \
>     --log-opt max-file=3 \
>     codalab/competitions-v2-compute-worker

The broker url corresponds to the queue I created on v2:
![image](https://user-images.githubusercontent.com/12370705/104155980-35e94280-5423-11eb-978b-c779fe700209.png)

The docker logs shows:

> /usr/local/lib/python3.8/site-packages/celery/platforms.py:800: RuntimeWarning: You're running the worker with supe
> ruser privileges: this is
> absolutely not recommended!
> Please specify a different user using the --uid option.
> User information: uid=0 euid=0 gid=0 egid=0
>   warnings.warn(RuntimeWarning(ROOT_DISCOURAGED.format(
>  
>  -------------- compute-worker@c3839e9a21b5 v4.4.0 (cliffs)
> --- ***** ----- 
> -- ******* ---- Linux-5.4.0-1025-gcp-x86_64-with-glibc2.2.5 2021-01-11 07:32:06
> - *** --- * --- 
> - ** ---------- [config]
> - ** ---------- .> app:         __main__:0x7f1ef93648e0
> - ** ---------- .> transport:   amqp://dd64f051-6cd3-4c4e-a646-ef5980276e1f:**@broker.codabench.org:9001/b348279e-a
> cb4-4178-8778-a8805f7dda3c
> - ** ---------- .> results:     disabled://
> - *** --- * --- .> concurrency: 1 (prefork)
> -- ******* ---- .> task events: OFF (enable -E to monitor tasks in this worker)
> --- ***** ----- 
>  -------------- [queues]
>                 .> compute-worker   exchange=compute-worker(direct) key=compute-worker
>                 
> [tasks]
>   . compute_worker_run
> [2021-01-11 07:32:07,130: ERROR/MainProcess] consumer: Cannot connect to amqp://dd64f051-6cd3-4c4e-a646-ef5980276e1
> f:**@broker.codabench.org:9001/b348279e-acb4-4178-8778-a8805f7dda3c: [Errno 104] Connection reset by peer.
> Trying again in 2.00 seconds...
> [2021-01-11 07:32:09,421: ERROR/MainProcess] consumer: Cannot connect to amqp://dd64f051-6cd3-4c4e-a646-ef5980276e1

Please help with this!

**Comment by ckcollab** [comment]
Specifically: I think you need `BROKER_USE_SSL=True`, the [wiki](https://github.com/codalab/competitions-v2/wiki/Compute-Worker-Management---Setup) had some nice instructions using a `.env` to configure it

**Comment by NehzUx** [comment]
Thanks for the update. I have gone through the updated instructions and succeeded with CPU queue setting up. After I submit a job, I can see from docker logs that this job is picked up.

However, I failed with GPU queue setup. After I followed the instructions, and run the command of "nvidia-docker", I can see from docker logs below:

> [2021-01-19 13:05:16,899: INFO/MainProcess] Connected to amqp://dd64f051-6cd3-4c4e-a646-ef5980276e1f:**@broker.coda
> bench.org:9001/8e373353-3d07-4ea1-8fe3-c773a6277a56
> [2021-01-19 13:05:17,735: INFO/MainProcess] mingle: searching for neighbors
> [2021-01-19 13:05:21,157: INFO/MainProcess] mingle: all alone
> [2021-01-19 13:05:23,479: INFO/MainProcess] compute-worker@14ff227091cb ready.

But after I submit a job, this queue remains this status and doesn't pick the job for processing as what I observe from CPU queue.

Thus, I have to reopen this issue and thanks a lot @ckcollab @Logan-Ruf  for checking this! GPU queue setting up is blocking us.

The command I used was:

> nvidia-docker run \
>     -v /home/zhxu_cs_gmail_com/codabench/cache:/codabench \
>     -v /var/run/docker.sock:/var/run/docker.sock \
>     -v /var/lib/nvidia-docker/nvidia-docker.sock:/var/lib/nvidia-docker/nvidia-docker.sock \
>     -d \
>     --env-file .env \
>     --restart unless-stopped \
>     --log-opt max-size=50m \
>     --log-opt max-file=3 \
>     codalab/competitions-v2-compute-worker:nvidia 

My '.env' file:

> 
> BROKER_URL=pyamqp://dd64f051-6cd3-4c4e-a646-ef5980276e1f:1dc2fa65-fd93-4260-bab2-0f39d6917030@broker.codabench.org:9001/8e373353-3d07-4ea1-8fe3-c773a6277a56
> HOST_DIRECTORY=/home/zhxu_cs_gmail_com/codabench/cache
> BROKER_USE_SSL=True

**Comment by ckcollab** [comment]
The nvidia worker may need an update, let me push a new build and lmk the results. Make sure your docker container is using this SHA version of the image: `e38de654879042edf85dc9b08766da624dd928537730d22514ceda5e7cacfec7`

**Comment by NehzUx** [comment]
OK, tell me when the nvidia-docker is updated and I will test again. Thanks a lot @ckcollab

**Comment by ckcollab** [comment]
It's updated.

**Comment by NehzUx** [comment]
This worked! Many thanks!

**Comment by NehzUx** [comment]
@ckcollab Hi Eric, do you put somewhere the dockerfile of the updated dockers (CPU and GPU both versions)?

**Comment by ckcollab** [comment]
If I am understanding you, the `Dockerfiles` are at the root level of the repository:

https://github.com/codalab/competitions-v2/blob/develop/Dockerfile.compute_worker

https://github.com/codalab/competitions-v2/blob/develop/Dockerfile.compute_worker_gpu

**Comment by NehzUx** [comment]
Yes, this is what I mean. But it seems not updated. You mentioned 1 week ago that you have updated the (gpu) docker. Could you update both dockerfiles (if any modification)? Many thanks!

> It's updated.

**Comment by ckcollab** [comment]
The docker file pulls in many files to make a "computer worker image." The file I changed was probably `worker.py`

You can see the version history and such here:
https://hub.docker.com/r/codalab/competitions-v2-compute-worker

**Comment by caizhongang** [comment]
@ckcollab
Sorry to trouble you.
I have followed exactly all steps of @NehzUx but the GPU worker does not work,
When I make a submission at the Codalab website, the status of the submission turns to 'fail' (without any error message) and nothing happens at the GPU worker:

``
[2021-06-23 13:23:44,075: INFO/MainProcess] Connected to amqp://ff72bcc5-5d8a-4b63-99b5-fa7c42be6e2c:**@competitions.codalab.org:5671/fa343a86-5f71-4a2a-9e56-a85d438d8c08
[2021-06-23 13:23:45,132: INFO/MainProcess] mingle: searching for neighbors
[2021-06-23 13:23:49,025: INFO/MainProcess] mingle: all alone
[2021-06-23 13:23:51,862: INFO/MainProcess] compute-worker@99ed58e99a33 ready.
``

I have done the following checks:
The version of the docker image is indeed ``e38de654879042edf85dc9b08766da624dd928537730d22514ceda5e7cacfec7``
The correct queue is used in the competition. I have checked with the competition-v1 docker image that the queue indeed works.

My ``.env`` file:
``
BROKER_URL=pyamqp://ff72bcc5-5d8a-4b63-99b5-fa7c42be6e2c:030ab285-20b1-459f-8b2c-763d4a13c381@competitions.codalab.org:5671/fa343a86-5f71-4a2a-9e56-a85d438d8c08
BROKER_USE_SSL=False
HOST_DIRECTORY=/tmp/codalab
``

My Docker command:
``
nvidia-docker run \
    -v /tmp/codabench:/codabench \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -v /var/lib/nvidia-docker/nvidia-docker.sock:/var/lib/nvidia-docker/nvidia-docker.sock \
    -d \
    --name gpu_worker \
    --env-file .env \
    --restart unless-stopped \
    --log-opt max-size=50m \
    --log-opt max-file=3 \
    codalab/competitions-v2-compute-worker:nvidia
``

Thank you in advance!
